client/docs/service_groups.md

```diff
@@ -0,0 +1,160 @@
+<!---
+Licensed under Creative Commons Attribution 4.0 International License
+https://creativecommons.org/licenses/by/4.0/
+--->
+
+# Service Groups #
+
+Service groups provide a means to simplify and reuse specification of
+services, resources and policies for creating new contracts.
+
+Contract creation requires specification of enclave, provisioning, and
+stoage services; preferred or optimized services; and storage and
+replication policies. Rather than provide the information separately
+for every contract created, service groups define collections of
+services and policies for using those services that can be used to
+simplify contract creation.  For example, a service group for testing
+and developing new contracts can be created from services provided
+locally. Another group can be defined to for final production
+deployment.
+
+Service groups can be easily defined for all of the services
+associated with a given site (see the various `create_from_site`
+subcommands below). They can also be created for services provided by
+multiple sites (for greater resiliency and availability).
+
+All service groups have a name and a list of services (generally
+represented by the names or URLs for the services defined in the
+service endpoint database). Each type of service group (`eservice`,
+`pservice`, and `sservice`) has additional unique attributes that
+describe policies that accompany the group. For example, enclave
+services have a preferred service that identifies a place more likely
+to have contract state pre-cached.
+
+While the name space for each type of service group is independent
+(that is, you can have an enclave service group with the same name as
+a provisioning service group), for convenience we often use the same
+name for groups of different types when we use them together. For
+example, we may name enclave, provisioning and storage service groups
+`develop` because we always use them together for contract
+development.
+
+While groups can be specified for any operation that creates a
+contract, if no group is specified, then a group called 'default' will
+be used. That group must be created.
+
+**NOTE:** Services included in service groups must be stored in the
+service endpoint database (see `pdo-service-db` command). Generally
+this means that service endpoint must be added (usually by importing a
+site configuration file) before groups can be created.
```

Comment:
For creating and working with local groups and exporting them this makes sense. But is that also true for the `import` command?   In that context it would seem useful to also (help to) update the service-db? Or is that handled at a higher level, e.g., in scripts importing contract bundles? Actually, from further below you seem to expect there is also a corresponding export/import from the endpoint-database: do we have a way to project the export for the end-point db covering just what is needed for this service group (e.g., i might have lots of other endpoints which i might not want to export (e.g., localhost ones would be meaningless)

Reply:
not sure i understand what you're asking. 1) i should add the check for existing services in the group import operation in the data wrapper (or... i go back and forth on this, the import should just be moved into the command). 2) the endpoint database does have an import/export. import/export is a full database option (unless the merge switch is used on import). 3) i'm not sure why we would need a partial import/export? (the file is just toml so you can easily edit it if you really don't want a bunch of extra stuff in there).

Reply:
moved the check for database existence into the actual store operation so it should be applied for all adds. there is a new verify() operation on groups that can check to make sure the URLs still exist in the db, but i'm not going to go any further for enforcing data integrity at this time.

Reply:
The use-case i had in mind (and maybe not intended to be addressed) that user A preparing ontract and as part of that creates this service group. At the point he would convince others, say party B, to join i expected that as part of the contract artifact transported would also be this service group and related end-point information.  Party B already might have some end-points and wants for sure only import end-point info related to the really necessary services, yet Party A's end-point database also might have end-points from development service groups which in best case Party B doesn't want to see and in worst-case might change existing (e.g., localhost) end-points Party B has ?   Hence the "projection" of end-points related to the service group.  Of course this can all be done manually but i thought the point of this here is to make it easier?

Reply:
the service group is already part of the contract file. look in the .pdo files in ${data}/__contract_cache__

Reply:
the second part was what where i was heading at ....

Reply:
Suggest we address that (very good) suggestion in a separate PR.